### DISCUSSION TITLE
Using BERTopic's topics_over_time with Image Data

### DISCUSSION THREAD

**Discussion by owwdesilva** [MaartenGr/BERTopic#2083] [Q&A]
Hi @MaartenGr, Hi All

Use of the BERTopic model for analyzing topics over time, specifically when dealing with sequences of images. I have trained a topic model using image data only, but the current implementation of topics_over_time seems to only accept documents and timestamps as input: `topics_over_time = topic_model.topics_over_time(docs, timestamps)`

Given that my model is based on image data, how can I adapt or use the topics_over_time functionality for a sequence of images? Any guidance or examples on how to approach this. Thanks!

**Comment by MaartenGr** [comment]
That is currently not possible since the procedure of `.topics_over_time` uses the c-TF-IDF representations of (sub)sets of documents to calculate the new representations. What you could do is attempt to caption your images and use that but as of now there is no support for images here. 

It is technically possible to do something like this by embedding and captioning the images again but that would require significant changes to the underlying code I'm afraid.

**Reply by owwdesilva** [reply]
Hi @MaartenGr. Thanks much for the reply. I'll explore captioning images as a workaround for now. Thanks again.
